### PR TITLE
The solution of the 'credit card' problem

### PR DESCRIPTION
**Pull request by koryun23**

### CODE REVIEW

File: src/test/com/aca/homework/week8/credit/card/PaymentServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package com.aca.homework.week8.credit.card;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+
+class PaymentServiceTest {
+    PaymentService testSubject;
```
Comment (gevorgyanarman) [inline src/test/com/aca/homework/week8/credit/card/PaymentServiceTest.java:9]: let's make this private

File: src/test/com/aca/homework/week8/credit/card/PaymentServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    @BeforeEach
+    public void setUp(){
```
Comment (gevorgyanarman) [inline src/test/com/aca/homework/week8/credit/card/PaymentServiceTest.java:12]: please format the code (ctrl +alt+ shift +L)

File: src/test/com/aca/homework/week8/credit/card/PaymentServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                testSubject.pay(creditCard, 150);
+            }
+        });
+    }
+
+    @Test
+    public void testWhenBalanceIs100AmountIs30Call4Times(){
+        CreditCard creditCard = new CreditCard(100, 1);
+        int countOfPayments = 0;
+        while(countOfPayments < 3){
```
Comment (gevorgyanarman) [inline src/test/com/aca/homework/week8/credit/card/PaymentServiceTest.java:31]: just call the pay method 3 times without the lopp

File: src/main/com/aca/homework/week8/credit/card/CreditCard.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package com.aca.homework.week8.credit.card;
+
+public class CreditCard {
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week8/credit/card/CreditCard.java:3]: the txt file is missing

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

**Comment by koryun23** [comment]
done

File: src/main/com/aca/homework/week8/credit/card/CreditCard.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    public int getNumber() {
+        return number;
+    }
+
+    public void setBalance(double balance) {
+        this.balance = balance;
+    }
+
+    public void setNumber(int number) {
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week8/credit/card/CreditCard.java:24]: the credit card number can not be changed. please remove it

File: src/main/com/aca/homework/week8/credit/card/InsufficientBalanceException.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package com.aca.homework.week8.credit.card;
+
+public class InsufficientBalanceException extends RuntimeException{
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week8/credit/card/InsufficientBalanceException.java:3]: please format the code

File: src/main/com/aca/homework/week8/credit/card/PaymentService.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package com.aca.homework.week8.credit.card;
+
+public class PaymentService {
+    public void pay(CreditCard creditCard, double amountToBePaid){
+        if(creditCard.getBalance() < amountToBePaid){
+            throw new InsufficientBalanceException("\n"+creditCard.toString() + "\ncould not transfer $" + amountToBePaid + ", because the balance($" + creditCard.getBalance() + ") is less than the amount to be paid.");
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week8/credit/card/PaymentService.java:6]: please move the exception text template to the exception constructor

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

**Comment by koryun23** [comment]
done

File: src/main/com/aca/homework/week8/credit/card/CreditCard.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        this.balance = balance;
+        this.number = number;
+    }
+
+    public double getBalance() {
+        return balance;
+    }
+
+    public void setBalance(double balance) {
+        this.balance = balance;
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week8/credit/card/CreditCard.java:17]: please add validations, someone can set negative balance

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]
please do the minor fix and merge the pull request

**Comment by koryun23** [comment]
added the validation and a test for that

**Comment by koryun23** [comment]
added the validation and a test for that

**Pull request merged**